daily-challenges/2024-09/17/min240917.spec.ts

```diff
@@ -0,0 +1,88 @@
+/* 
+Javascript
+Đề bài:
+Số nguyên tố là một số tự nhiên lớn hơn 1 và chỉ chia hết cho 1 và chính nó. Trong bài tập này, bạn sẽ viết một hàm để kiểm tra xem một số có phải là số nguyên tố hay không.
+
+Một số nguyên tố là số chỉ có đúng hai ước là 1 và chính nó. Để kiểm tra một số n có phải số nguyên tố hay không, bạn cần kiểm tra xem n có chia hết cho bất kỳ số nào từ 2 đến căn bậc hai của n không. Nếu có, nó không phải số nguyên tố
+
+### Yêu cầu:
+- Viết một hàm JavaScript có tên `isPrime` để kiểm tra xem một số có phải là số nguyên tố không.
+- Nếu số là số nguyên tố, in ra "Số này là số nguyên tố". Nếu không phải, in ra "Số này không phải là số nguyên tố". 
+*/
+
+function isPrime(number: number) {
+    if (number < 2) {
+        console.log(`"Số này không phải là số nguyên tố"`);
+    }
+    else if (number == 2) {
+        console.log(`"Số này là số nguyên tố"`);
+    }
+    else if (number % 2 == 0) {
+        console.log(`"Số này không phải là số nguyên tố"`);
+    }
+    else if (number > 2) {
+        for (let i = 0; i <= number - 1; i += 2) {
```

Comment:
Có thể tối ưu hơn bằng cách: chạy từ 2 tới căn bậc 2 của number: Math.sqrt(number) 
```suggestion
        for (let i = 2; i <= Math.sqrt(number; i += 2); i++) {
```

Reply:
updated